src/libraries/System.IO.FileSystem/tests/File/EncryptDecrypt.Windows.cs

```diff
@@ -0,0 +1,85 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.DotNet.XUnitExtensions;
+using System.Diagnostics;
+using System.Diagnostics.Eventing.Reader;
+using System.Security;
+using System.ServiceProcess;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.IO.Tests
+{
+    public partial class EncryptDecrypt
+    {
+        partial void LogEFSDiagnostics()
+        {
+            try
+            {
+                using var sc = new ServiceController("EFS");
+                _output.WriteLine($"EFS service is: {sc.Status}");
+                if (sc.Status != ServiceControllerStatus.Running)
+                {
+                    _output.WriteLine("Trying to start EFS service");
+                    sc.Start();
```

Comment:
So after we run the test and it fails, we check whether the EFS service is running and if it's not, we are restarting it.

@danmoseley perhaps we should try to start running it before we run the test? or run the test only when the service is already running similar to [local file share tests](https://github.com/dotnet/runtime/blob/579ef5be313aea703818e2225b70cb7714a90cf3/src/libraries/System.IO.FileSystem/tests/FileStream/FileStreamConformanceTests.Windows.cs#L82-L99) ?

Reply:
Applied both feedback.